install_files/ansible-base/roles/grsecurity/tasks/paxctl.yml

```diff
@@ -0,0 +1,36 @@
+---
+- name: Install paxctl.
+  apt:
+    pkg: paxctl
+    state: present
+  tags:
+    - apt
+
+- name: Check paxctl headers on grub binaries.
+  command: paxctl -vQ {{ item }}
+  # Read-only task, so don't report changed
+  changed_when: false
+  # The first time this task is run, the grub binaries won't have
+  # their headers converted yet, which will cause paxctl to exit 1.
+  # We'll catch that error and respond accordingly in the next task.
+  failed_when: false
+  register: paxctl_grub_header_check
+  with_items:
+    - /usr/sbin/grub-probe
+    - /usr/sbin/grub-mkdevicemap
+
+- name: Adjust paxctl headers on grub binaries.
+  command: paxctl -Cpm {{ item.item }}
+  with_items: paxctl_grub_header_check.results
+  # The desired flags should include:
+  #   - p NOPAGEEXEC
+  #   - m NOMPROTECT
+  #   - x NORANDEXEC
+  #   - e NOEMUTRAMP
+  # After running `paxctl -Cpm` in VirtualBox, all four of those flags are set.
+  # However, the current paxctl command only enforces p and m.
+  # The x and e flags are present by default, but can be removed via
+  # `paxctl -z <binary>`. TODO: We may want to update the command
+  # to set -Cpmxe explicitly. Warrants further testing on hardware.
```

Comment:
For what it's worth, the default paxctld configuration file sets flags for the following binaries:
```
/usr/bin/grub-script-check      E
/usr/bin/grub-bios-setup        E
/usr/sbin/grub-mkdevicemap      E
/usr/sbin/grub-probe            E
```
However, this is enabling EMUTRAP instead of disabling MPROTECT. I would like to learn more about the distinctions (trampolines, etc.) and why this is recommended. I've found in testing that these GRUB utilities don't work on that that configuration, ex. installing a new kernel will fail on the post-install hooks, but they do run without any PaX flags. In other words, the flags recommended by paxctld.conf cause breakage. Same for Python - enabling EMUTRAP is not enough, although this is what's recommended instead of going all out on disabling MPROTECT. We do still need to get to the bottom of this.